stacks-common/build.rs

```diff
@@ -1,8 +1,64 @@
 use std::path::Path;
+use std::process::Command;
 use std::{env, fs};
 
 use toml::Value;
 
+fn current_git_hash() -> Option<String> {
+    if option_env!("GIT_COMMIT") == None {
+        let commit = Command::new("git")
```

Comment:
Could you refactor this a bit, creating a new function, `run_git_command` or something, that all three of these new functions could use?